en/developer_guide/bbcode.rst

```diff
@@ -0,0 +1,153 @@
+BBCodeBundle to collect all services tagged with ``open_orchestra_bbcode.code_definitions`` and add
+the matching definition to the master definition set.
+
+Such a service must implement ``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionCollectionInterface``.
+This collection must be composed of elements implementing
+``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionInterface``. You can quickly generate
+definitions implementing that interface using the OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionFactory.
+This factory can be accessed via the service named ``open_orchestra_bbcode.definition.factory``
+
+To generate a new simple definition with this factory, call the method ``create`` with the following
+parameters:
+
+- ``$tag``: the BBCode tag, ie ``b`` for the tag [b]
+
+- ``$html``: the html representation of the tag, ie ``<strong>{param}</strong>`` for the [b]tag
+
+- ``$use_option``: whether the tag uses option or not
+
+So to create a bold tag definition object, you can write:
+
+.. code-block:: php
+
+    $definition = $container->get('open_orchestra_bbcode.definition.factory')->create('b', '<strong>{param}</strong>');
+
+And to create a header tag definition:
+
+.. code-block:: php
+
+    $definition = $container->get('open_orchestra_bbcode.definition.factory')->create('h', '<h{option}>{param}</h{option}>', true);
+
+You can now add this definition to a class implementing
+``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionCollectionInterface``, and make it a Symfony
+service tagged with ``open_orchestra_bbcode.code_definitions``.
+
+Create a complex BBCode tag via service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Maybe the tag you want to create requires a more complex generation. For instance a tag ``youtube``
+could check in its parameter a valid youtube id and then render an iframe if found.
+
+A specific PHP code needs to be executed to render the tag. To do this your definition could implement
+``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionInterface``.
+The html version of the tag is rendered by the method getHtml, so you can code here the specific
+logic of your tag. This method receives a BBcodeElementNode representing the BBCode tag. Here is
+what could look like the getHtml method for the ``youtube`` tag:
+
+.. code-block:: php
+
+    public function getHtml(BBcodeElementNode $el)
+    {
+        $content = '';
+        foreach ($el->getChildren() as $child) {
+            $content .= $child->getAsBBCode();
+        }
+        $foundMatch = preg_match('/v=([A-z0-9=\-]+?)(&.*)?$/i', $content, $matches);
+        if (!$foundMatch) {
+            return $el->getAsBBCode();
+        }
+        return '<iframe width="640" height="390" src="http://www.youtube.com/embed/'
+            . $matches[1] . '" frameborder="0" allowfullscreen></iframe>';
+    }
+
+Using a service to define your tag allows you to access the Symfony container and to use services
+to do a complex transformation.
```

Comment:
The definition doen't have to be a tagged service, the definitions collection must. With your comment, I understand that this point is not clear, so I rewrite the service part to be clearer.